frontend/src/core/runtime/runtime.ts

```diff
@@ -52,14 +56,14 @@ export class RuntimeManager {
       }
     }
 
-    // Move over window level parameters to the WebSocket URL
-    // if they are "known" query params.
-    for (const lookup in KnownQueryParams) {
-      const key = KnownQueryParams[lookup as keyof typeof KnownQueryParams];
-      const value = currentParams.get(key);
-      if (value !== null) {
-        baseUrl.searchParams.set(key, value);
+    for (const [key, value] of currentParams.entries()) {
+      if (
+        restrictToKnownQueryParams &&
+        !Object.values(KnownQueryParams).includes(key)
+      ) {
+        continue;
       }
```

Comment:
I included this in case other code paths do need to restrict to known query parameters.

Reply:
Got it, so just don't filter known query params, pass all through?